doc/rapidcheck.md

```diff
@@ -0,0 +1,84 @@
+# RapidCheck property-based testing for Bitcoin Core
+
+## Concept
+
+Property-based testing is experimentally being added to Bitcoin Core with
+[RapidCheck](https://github.com/emil-e/rapidcheck), a C++ framework for
+property-based testing inspired by the Haskell library
+[QuickCheck](https://hackage.haskell.org/package/QuickCheck).
+
+RapidCheck performs random testing of program properties. A specification of the
+program is given in the form of properties which functions should satisfy, and
+RapidCheck tests that the properties hold in a large number of randomly
+generated cases.
+
+If an exception is found, RapidCheck tries to find the smallest case, for some
+definition of smallest, for which the property is still false and displays it as
+a counter-example. For example, if the input is an integer, RapidCheck tries to
+find the smallest integer for which the property is false.
+
+## Running
+
+If RapidCheck is installed, Bitcoin Core will automatically run the
```

Comment:
Do we want to run these by default? This forces everyone to install rapidcheck. I think that is reasonable when we have a pre-configured way to easily install rapidcheck. AFAIK this isn't this case, unless I missed a PR.

Reply:
Currently, RapidCheck is auto-configured only if librapidcheck.a is detected, which IIUC doesn't force anyone to install it. I think this is outside the scope of the doc which addresses the current situation. FWIW I'm working on updating the RapidCheck Makefile to maybe/hopefully autobuild cleanly and can update this doc in that PR if needed.

Reply:
(Rapidcheck auto-configure as seen in configure.ac:L134-138 and 1280-1292)

Thanks for reviewing!